app/controllers/concerns/secure_headers_concern.rb

```diff
@@ -1,7 +1,21 @@
 module SecureHeadersConcern
   extend ActiveSupport::Concern
 
+  class_methods do
+    def apply_secure_headers_override(options = {})
+      return unless FeatureManagement.rails_csp_tooling_enabled?
+
+      before_action(options) do
+        policy = current_content_security_policy
+        policy.form_action *csp_uris
+        request.content_security_policy = policy
+      end
+    end
+  end
+
   def apply_secure_headers_override
```

Comment:
what if we kept this as a before action and did some more work in this method? instead of having both the class action that defines an anonymous filter, we can just use the named filter we already have?

Reply:
Ah yes, that would be smarter.

We originally tried doing it the way the rails guide suggested but had some weird scoping issues prevent us from being able to get to `csp_uris` so we wired in a `before_action`. In hindsight, makes more sense to just do it in the original before action.